kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabWirelessUi.java

```diff
@@ -1814,34 +1794,36 @@ public void onFailure(Throwable caught) {
                                 @Override
                                 public void onSuccess(List<GwtWifiChannelFrequency> freqChannels) {
 
-                                    TabWirelessUi.this.channelList.clear();
-
-                                    addAutomaticChannel(freqChannels);
+                                    updateChannelListValues(freqChannels);
 
-                                    freqChannels.stream().forEach(TabWirelessUi.this::addItemChannelList);
-
-                                    List<Integer> currentChannels = TabWirelessUi.this.activeConfig.getChannels();
+                                }
+                            });
+                }
+            });
+        }
 
-                                    if (currentChannels != null && !currentChannels.isEmpty()) {
-                                        int channel = TabWirelessUi.this.activeConfig.getChannels().get(0);
+    }
 
-                                        int selectedChannelIndex = getChannelIndexFromValue(channel);
-                                        int channelIndex = selectedChannelIndex == -1 ? 0 : selectedChannelIndex;
+    private void updateChannelListValues(List<GwtWifiChannelFrequency> freqChannels) {
 
-                                        TabWirelessUi.this.channelList.setSelectedIndex(channelIndex);
+        int selectedChannelValue = 0;
 
-                                    }
+        if (TabWirelessUi.this.channelList.getItemCount() != 0) {
+            selectedChannelValue = this.channelList.getSelectedIndex();
+            this.channelList.clear();
```

Comment:
It is best to clear outside this if, otherwise you will end up duplicate elements when changing radio mode.